core/src/main/java/org/apache/iceberg/ManifestListWriter.java

```diff
@@ -70,12 +95,31 @@ public long length() {
     return writer.length();
   }
 
+  public ManifestListFile toManifestListFile() {
```

Comment:
Rather than having `SnapshotProducer` handle encryption, I've moved that logic to this class. This is similar to the pattern used for `ManifestFile` except that the `keyMetadata` also needs to be encrypted. To handle that, I've updated this to use the `EncryptionManager`.